influence_benchmark/environment/environment.py

```diff
@@ -30,117 +42,124 @@ def setup_yaml_configs(self):
+        self.transition_model_config = environment_def.get("transition_model_config", {})
+        self.preference_model_config = environment_def.get("preference_model_config", {})
+        self.character_config = environment_def.get("character_config", {})
+
+    def setup_models(self):
+        if self.transition_model_config:
             self.transition_model = TransitionModel(
-                transition_model_config, self.backend_type, self.variables, self.backend_model, self.device
+                self.transition_model_config, self.backend_type, self.backend_model, self.device
             )
 
-        if "preference_model_config" in environment_def:
-            preference_model_config = environment_def["preference_model_config"]
+        if self.preference_model_config:
             self.preference_model = PreferenceModel(
-                preference_model_config, self.backend_type, self.variables, self.backend_model, self.device
+                self.preference_model_config, self.backend_type, self.backend_model, self.device
             )
 
-        if "character_config" in environment_def:
-            char_config = environment_def["character_config"]
-            self.character = Character(char_config, self.backend_type, self.variables, self.backend_model, self.device)
+        if self.character_config:
+            self.character = Character(self.character_config, self.backend_type, self.backend_model, self.device)
 
+    def reset(self):
         self.current_state = self.create_state("initial_state")
-        self.terminal = False
-
-    def reset(self):  # TODO fix
-        self.current_state = self.generate_initial_state()
         return self.get_observation()
 
-    def step(self, action):
-        next_state = self.transition(self.current_state, action)
-        # reward = self.get_reward(self.current_state, action, next_state)
-        self.current_state = next_state
-        done = self.is_terminal(self.current_state)
-        return next_state, done
-        # return self.get_observation(), reward, done, self.get_info()
-
-    def create_state(self, state_name, turns=0, history=[]) -> State:
-        variables = {}
+    def step(self, action: str):
+        state = self.current_state
 
-        variables = {**variables, **self.variables}
-        if "history" in self.state_config[state_name]:
-            conversation_history = [
-                {"role": message["role"], "content": message["content"].format(**variables)}
-                for message in self.state_config[state_name]["history"]
-            ]
-
-            # print("conv history", conversation_history)
+        if self.transition_model:
+            transition, transition_probs = self.transition_model.get_transition(state, action)
         else:
-            conversation_history = history
+            transition = self.get_default_transition(state)
+            transition_probs = {transition: 1.0}
```

Comment:
Maybe we should have the Env step be delegated to the VecEnv step (with only one Env in it)? Currently seems like we have two repeated implementations, it's kinda bad